x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/StoreKeyConfig.java

```diff
@@ -79,6 +79,20 @@ X509ExtendedKeyManager createKeyManager(@Nullable Environment environment) {
         try {
             KeyStore ks = getStore(ksPath, keyStoreType, keyStorePassword);
             checkKeyStore(ks);
+            // TBD: filte out only http.ssl.keystore
+            ArrayList<String> aliases = Collections.list(ks.aliases());
+            if (aliases.size() > 1) {
```

Comment:
if you'd filter only key entries before you check for the alias list size, then you don't have to iterate the list in the case where a keystore contains i.e. a PrivateKeyEntry and a TrustedCertificateEntry. As we said, we only want to be doing filtering _iff_ we have more than one PrivateKeyEntry objects. This code will be doing filtering even when we have one PrivateKeyEntry and something else. 

This might be more efficient to do with a stream , while keeping the readability.

Reply:
Good point, changed to checking for the keys first.
Don't like it with stream though, the exception handling looks ugly (isKeyEntry() can throw)